src/gc/gcinterface.ee.h

```diff
@@ -219,6 +219,51 @@ class IGCToCLR {
     // or a server GC thread.
     virtual
     bool WasCurrentThreadCreatedByGC() = 0;
+
+    // Given an object, if this object is an instance of `System.Threading.OverlappedData`,
+    // and the runtime treats instances of this class specially, traverses the objects that
+    // are directly or (once) indirectly pinned by this object and reports them to the GC for
+    // the purposes of relocation and promotion.
+    //
+    // Overlapped objects are very special and as such the objects they wrap can't be promoted in
+    // the same manner as normal objects. This callback gives the EE the opportunity to hide these
+    // details, if they are implemented at all.
+    //
+    // This function is a no-op if "object" is not an OverlappedData object.
+    virtual
+    void WalkOverlappedObjectForPromotion(Object* object, ScanContext* sc, promote_func* callback) = 0;
```

Comment:
I would call it `AsyncPinned`. AsyncPinned better describes the primitive, and it is what we call it on other places in the gcee interface (e.g. there is RelocateAsyncPinnedHandles method).

Reply:
will do!